### ISSUE TITLE
Explosive slugs are hilariously underpowered

### ISSUE THREAD

**Issue by Terrorforge** [CleverRaven/Cataclysm-DDA#53179]
### Describe the bug

Explosive slugs are rare and described as "damaging", but in reality they deal less damage than a pistol shot. It seems to be somewhere around ~18 to the primary target, with the direct hit and explosion both dealing less than 10 damage.

### Steps to reproduce

1. Give yourself any shotgun and some explosives slugs
2. Spawn a zombie
3. Shoot it
4. Note that it barely takes any damage

### Expected behavior

These super rare, nasty-sounding grenade rounds should probably do more damage than standard ammunition. Of course, buckshot already does a lot of damage these days, so for these to be as cool as they seem they would need to either have a proper AoE effect, or deal a frankly obscene amount of damage to the primary target.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 661b2fd [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

Here's some video and description of the real thing in action: https://anthonysfirearmwarehouse.com/exploding-shotgun-ammo-meet-the-frag-12/

**Comment by catdach** [comment]
I feel you man, I really do. Explosive slugs are very underwhelming and could probably use a buff, especially considering how rare they are. Unfortunately the whole situation is a little complicated. (I spent about 5-6 hours down the rabbit hole so this might get a bit long winded)

TLDR: I might make a PR to buff them slightly, but they are probably going to remain extremely weak (probably going to request an audit on explosives generally)

Ok, so how much damage should they do? Right now they are coded to be 10 bullet damage which isn't much, but the point is to explode not be a regular slug so that can probably remain the same. I'll just buff the explosion damage right? well...

In CDDA explosives are each given their own unique explosion data in `ammo_effects.json` explosive slugs use the following explosion data.

    "id": "EXPLOSIVE_HESHOT",
    "type": "ammo_effect",
    "explosion": { "power": 60 }

Ok, so, do I just increase `power`? Um...well I could, but lets look at another explosive for a sec.

    "id": "EXPLOSIVE_m430a1",
    "type": "ammo_effect",
    "explosion": { "power": 72, "shrapnel": { "casing_mass": 250, "fragment_mass": 0.08 } }

This is the explosive data for the `40x53mm M430A1 HEDP` which is one of the more powerful explosive ammunition in the game (ignore the `shrapnel` part for now, Ill get to that later).  Note that the power is 72. Clearly the data for explosives is a little more complex than `power = damage` so it might be time to look at some official guidelines for explosive data. the following is from `doc/JSON_INFO.md`.



`"explosion":{                               // Physical explosion data`

    "power": 10,                             // Measure of explosion power in grams of TNT equivalent explosive, affects damage and range.
    "distance_factor": 0.9,                  // How much power is retained per traveled tile of explosion. Must be lower than 1 and higher than 0.
    "max_noise": 25,                         // Maximum amount of (auditory) noise the explosion might produce.
    "fire": true,                            // Should the explosion leave fire
    "shrapnel": 200,                         // Total mass of casing, rest of fragmentation variables set to reasonable defaults.
    "shrapnel": {
        "casing_mass": 200,                  // Total mass of casing, casing/power ratio determines fragment velocity.
        "fragment_mass": 0.05,               // Mass of each fragment in grams. Large fragments hit harder, small fragments hit more often.
        "recovery": 10,                      // Percentage chance to drop an item at landing point.`

OKAY! So now I can throw in some real world data, add the shrapnel data, and call it a day, right? 
Two problems with that:
1. Data on IRL explosive slugs is limited at best and outright restricted from the public at worse
2. The data for explosions is in need of an audit so I cant even scale down something that exists in game or even use existing items referentially for a ballpark estimate.

Problem 1:
My best guess is that the explosive slug is based on the [FRAG-12](https://en.wikipedia.org/wiki/FRAG-12) the most useful info on the wikipedia page is;

> The shell uses a 50 mm (2.0 in) long metal body filled with 3.4 g (0.12 oz) of composition A5.

Other information that I could find was either speculative or a literal advertisement for the ammunition (which weren't very informative anyway). I would file the link posted [in the request](https://anthonysfirearmwarehouse.com/exploding-shotgun-ammo-meet-the-frag-12/) under the 'advertisement' category; while the videos are _moderately_ impressive, the first one is zoomed in, shot in slow-mo, and really doesn't give any information other than the visual impressiveness of the fireball (aka it looks pretty). the second video is a bit more revealing, It's hard to tell anything about the target they're shooting at but those expositions look tiny. Even if they were more informative, I doubt they're unbiased considering the are [literally on the official website where you can buy the ammo](frag12.com/home)

As a side note, the official website has a "datasheet" section but it requires a password to access it so I'm not getting anything from there.(why they even have it on the site I'll never know)

Bottom line, the ammunition is relatively new, _probably_ have limited practicality, unavailable to most civilians, and are of questionable levels of interest to most military and police forces. Information on the thing is going to be as rare as the ammo itself.

Problem 2:
remember that Wikipedia line earlier? "composition A5" is the same stuff [used in M430A1](https://bulletpicker.com/pdf/TM%2043-0001-28,%20Artillery%20Ammunition.pdf#page=597) (I found that link through a google search, please don't end me to jail). The M430A1 uses 32 g of A5 under "filler and weight" which is under "projectile" how does A5 translate to TNT so I can finally put some number in `power` that makes sense? **I HAVE NO IDEA** do I just scale power down to 6 then? 

Before that lets look at the some numbers that are already in the power stat. 

dynamite: 938
grenade: 240
explosive slug: 60
M430A1: 72
M433: 42

Oh yeah, the M433 is also in that document that I linked before. You know how much A5 it uses? **45 g**. I...*sigh* wtf man? There are the other properties like shrapnel and distance factor that help to keep these things balanced but that doesn't change the fact that these number seem way, WAY off. I might give the slug some shrapnel data to temporarily make it better. Oh, the numbers for shrapnel seem like they're wrong too btw (don't get me started). 

TLDR 2: I'm going to request an audit on explosives if there isn't one happening already.

**Comment by Terrorforge** [comment]
One thing about adding shrapnel: these things have a maximum range of 12 tiles, and can't be peek-thrown like grenades. I've had shrapnel travel over 20 tiles, so I would be concerned that these could become very hazardous to the player. Proceed with caution.

I looked at the FRAG-12 stuff, too, and, well... any way you slice it, 3.4g of RDX is something like 2% of the explosives in a [M67 hand grenade](https://en.wikipedia.org/wiki/M67_grenade). And if you look at the video, they don't actually seem to have very much area effect. There's a man-sized fireball, and they punch a big hole, but that seems to be about it.

Now, we can of course fudge a little bit. Explosive slugs are found in the same context as near-future Rivtech guns, so there's some leeway. But you'd need a _lot_ of leeway to make something that fits down a shotgun bore have an explosive effect remotely comparable to a grenade.

The alternative that comes to me is to essentially make them super-slugs. Give them high damage and armor penetration and say they're shaped charges that blow/burn a big hole in whatever they hit. Keep the weaksauce explosion for flavor. I don't love this option, because I _want_ to be shooting grenades out of my 12-gauge, but it's the only one I can kind of square with reality.

**Comment by catdach** [comment]
Ok so, I found the [conversion ](https://en.wikipedia.org/wiki/TNT_equivalent#Relative_effectiveness_factor)for Composition A-5 to TNT, it's 1.6. so realistically the `power` would be at 5.44 grams of TNT. I'm not going to nerf it because I don't even know if the game can handle that small of an explosion lol. 

Reality is that your shotgun isn't going to replace an _actual_ grenade launcher. explosive shotgun shells are more of a novelty than anything else. It is still an explosion, which has potential to be useful if you shoot it at a gas tank, or pile of dynamite, or pile of loose ammo, or whatever; It's just not something for directly killing a target. 

On the rarity side of things...well not everything is rare because it's _good_, sometimes it rare for the opposite reason.
(most civilians cant get it and most militaries don't really want it, hence rarity)

**Comment by PatrikLundell** [comment]
I've only read the above, and not followed any links, but there's one (potential) factor I haven't seen discussed: namely WHERE the explosion takes place. It's a major difference between triggering a hand grenade on top of an armor covered belly and drilling a hole through the armor into said belly and blow up an explosive on the inside (I wouldn't want to be the owner of said belly in either case, though). If the explosion is delayed until the charge is inside the target the effect will be a lot larger than if it explodes on impact (anti tank weapons have gone through that kind of development, where a plasma drills a hole through the plating and the charge explodes inside the tank, rather than on impact). The reason is that essentially all of the energy go into the target, rather than the majority going into the surroundings and the armor.
I don't think the current system supports these considerations, so the fudge would probably be to increase the power by a large amount. However, that wouldn't handle the case where there actually wasn't any penetration into consideration.

**Comment by Alex-Folts** [comment]
> I looked at the FRAG-12 stuff, too, and, well... any way you slice it, 3.4g of RDX is something like 2% of the explosives in a [M67 hand grenade](https://en.wikipedia.org/wiki/M67_grenade). And if you look at the video, they don't actually seem to have very much area effect. There's a man-sized fireball, and they punch a big hole, but that seems to be about it.

Isnt this like expansive bullets, it kind of exploding when it inside the body to deal a lot of destruction?  I mean compared to expanding bullets, shotgun slug have higher calliber + active explosive inside, it should rip of large part of the zombie on hit. Ofc when you hit armor effect should be minimal(though it should damage armor itself). Maybe they just bugged?

**Comment by catdach** [comment]
> > I looked at the FRAG-12 stuff, too, and, well... any way you slice it, 3.4g of RDX is something like 2% of the explosives in a [M67 hand grenade](https://en.wikipedia.org/wiki/M67_grenade). And if you look at the video, they don't actually seem to have very much area effect. There's a man-sized fireball, and they punch a big hole, but that seems to be about it.
> 
> Isnt this like expansive bullets, it kind of exploding when it inside the body to deal a lot of destruction? I mean compared to expanding bullets, shotgun slug have higher calliber + active explosive inside, it should rip of large part of the zombie on hit. Ofc when you hit armor effect should be minimal(though it should damage armor itself). Maybe they just bugged?

Well...no, For a few reasons. I'm no ballistics expert but I'll explain the best I can. For regular bullets and shotgun slugs, stopping power can be measured in [Taylor knock-out factor (TKOF)](https://en.wikipedia.org/wiki/Taylor_knock-out_factor). The formula for TKOF is: `TKOF = bullet mass(grains) * muzzle velocity(feet per second) * diameter(in inches) * 1/7000` The TKOF of 5.56x45 is 5.6, The TKOF of 7.62x51 is 17.9. Expanding bullets reduce over penetration (the bullet going through the target), this ensures that none of the kinetic force of the bullet is lost and produces a larger wound at the cost slightly reduced mass. The Frag-12 is not that. As far I can tell, the frag-12 would much less propellant than a normal slug due to the size requirements of the explosive device (meaning worse muzzle velocity) AND It has less mass because active explosive and everything else that it needs to function is never going to be as dense as solid lead. Keep in mind that both of these numbers going down makes the TKOF go down exponentially (because the formula multiplies them together)

While the frag-12 may have some things in common with expanding bullets, they are both meant to reduce over penetration and produce a nasty wound, the frag-12 is the worse of the two. the only the way that the frag 12 reduces over pen is by having vastly reduced velocity (it doesn't have a choice, it doesn't expand and if it ever over pens before exploding then it would be worse than a normal slug), it's all in on the explosive to do the damage. And that explosive seems fairly unimpressive (as has been discussed).

shrapnel/fragmentation is a whole other can of worms but the bottom line is that it's really difficult to evaluate without more information (which as I've said previously is hard to find).

TLDR: All signs point to the explosive slug just being a really weak round; its mass is reduced to facilitate an explosive device, its muzzle velocity is reduced to prevent over penetration, the explosive agent is relatively unimpressive, and info on fragmentation is nonexistent.

**Comment by Terrorforge** [comment]
If we're not gonna make them useful, then at least the description needs to change. "Damaging", my ass
